Subscriber/CheckoutSubscriber.php

```diff
@@ -414,61 +161,51 @@ private function checkFirstCheckoutStep(Shopware_Controllers_Frontend_Checkout $
     private function shouldRedirectToStep2(Shopware_Controllers_Frontend_Checkout $subject): bool
     {
         $userData = $subject->View()->getAssign('sUserData');
-        if (!$userData['additional'] ||
-            !$userData['additional']['payment'] ||
-            $userData['additional']['payment']['name'] !== AdyenPayment::ADYEN_GENERAL_PAYMENT_METHOD) {
+        $source = (int) ($userData['additional']['payment']['source'] ?? null);
+        if (SourceType::load($source)->equals(SourceType::adyen())) {
```

Comment:
I created a separate issue #185 because this check seems to be causing an error on the checkout